src/MongoDB.Analyzer/Core/Builders/BuildersVariablesResolver.cs

```diff
@@ -0,0 +1,396 @@
+            {
+                isValidBuilderExpression = (semanticModel.GetSymbolInfo((declarator.Parent as VariableDeclarationSyntax).Type).Symbol as ITypeSymbol).IsBuilderDefinition() && !ContainsConditionalExpression(syntaxNode);
+            }
+            else if (syntaxNode is AssignmentExpressionSyntax assignment)
+            {
+                var leftHandSide = assignment.Left.TrimParenthesis();
+                isValidBuilderExpression = (leftHandSide is TupleExpressionSyntax tuple ? tuple.Arguments.Any(u => semanticModel.GetTypeInfo(u.TrimParenthesis()).Type.IsBuilderDefinition()) : semanticModel.GetTypeInfo(leftHandSide).Type.IsBuilderDefinition()) && !ContainsConditionalExpression(syntaxNode) && (assignment.Right is not AssignmentExpressionSyntax ? leftHandSide is IdentifierNameSyntax : true);
+            }
+
+            return isValidBuilderExpression;
+        }
+        
+        bool IsPartOfAccessExpression(IdentifierNameSyntax identifierName) => identifierName.Parent is MemberAccessExpressionSyntax || identifierName.Parent is ElementAccessExpressionSyntax;
+
+        IEnumerable<string> ScrapeVariables(AssignmentExpressionSyntax assignment) => assignment.Left.DescendantNodesAndSelf().OfType<IdentifierNameSyntax>().Where(n => !IsPartOfAccessExpression(n)).Select(n => n.Identifier.ValueText);
+    }
+
+    private static bool IsConditionalBlock(SyntaxNode syntaxNode) =>
+        syntaxNode switch
+        {
+            LocalFunctionStatementSyntax => false,
+            SwitchSectionSyntax or
+            SwitchExpressionArmSyntax or
+            ElseClauseSyntax or
+            ForStatementSyntax or
+            CommonForEachStatementSyntax or
+            DoStatementSyntax or
+            WhileStatementSyntax or
+            CatchClauseSyntax => true,
+            _ => syntaxNode.Parent is TryStatementSyntax || syntaxNode.Parent is IfStatementSyntax
+        };
+
+    private static void ProcessBlock(SyntaxNode syntaxNode, ResolutionContext resolutionContext)
+    {
+        var dirtyVariables = new HashSet<string>();
+        HashSet<SyntaxNode> nodesProcessed = null;
+
+        foreach (var childNode in syntaxNode.DescendantNodes(n => ShouldDescendIntoChildren(n)))
+        {
+            if (nodesProcessed.ContainsSafe(childNode.Parent))
+            {
+                continue;
+            }
+
+            if (IsConditionalBlock(childNode))
+            {
+                ProcessBlock(childNode, resolutionContext);
+                MarkNodeAsProcessed(childNode);
+            }
+            else if (TryProcessBuildersExpression(childNode, resolutionContext, dirtyVariables))
+            {
+                MarkNodeAsProcessed(childNode);
+            }
+        }
+
+        resolutionContext.CurrentVariablesValues.RemoveRange(dirtyVariables);
+
+        bool ShouldDescendIntoChildren(SyntaxNode syntaxNode) => !nodesProcessed.ContainsSafe(syntaxNode.Parent) && syntaxNode is not LocalFunctionStatementSyntax;
```

Comment:
Move the body into new line:
bool ShouldDescendIntoChildren(SyntaxNode syntaxNode) =>
     !nodesProcessed....

Reply:
Done